### PR TITLE
feat: `createWebWorkerHandler` for Cloudflare and Deno. Deprecates `createCloudflareHandler`

### PR DESCRIPTION
**Pull request by baoshan**
The [current implementation](https://github.com/octokit/oauth-app.js/blob/1c04f9b54e99c9c81db3cfa28510d6c39154e7b9/src/middleware/cloudflare/index.ts#L16) of `createCloudflareHandler` is not generic, preventing it from accepting an `OAuthApp` with `clientType: "github-app"`.

This PR also renames `createCloudflareHandler ` to `createWebWorkerHandler`. The same handler supports both Cloudflare workers and Deno. Documentation is updated to reflect the change.

### CODE REVIEW

**Comment by wolfy1339** [comment]
You seem to have included a commit which is not needed, as it's changes are already present on the repository [c632013](https://github.com/octokit/oauth-app.js/pull/287/commits/c632013ecd24d94a8d82b35186b978e90c809052) should be removed from the PR

File: package-lock.json [hunk 1]
```diff
@@ -3421,7 +3421,14 @@
       "version": "1.0.10",
       "resolved": "https://registry.npmjs.org/argparse/-/argparse-1.0.10.tgz",
       "integrity": "sha512-o5Roy6tNG4SL/FOkCAN6RzjiakZS25RLYFrcMttJqbdd8BWrnA+fGz57iN5Pb06pvBGvl5gQ0B48dJlslXvoTg==",
```
Comment (wolfy1339) [inline package-lock.json:3423]: Seems like you forgot some merge conflicts here

**Review by wolfy1339 — DISMISSED** [review]
Looks good to me!
Just remove the `package-lock.json` to avoid any conflicts

**Review by wolfy1339 — DISMISSED** [review]

File: src/middleware/handle-request.ts [hunk 2]
```diff
@@ -89,16 +89,13 @@ export async function handleRequest(
           `[@octokit/oauth-app] ${query.error} ${query.error_description}`
         );
       }
-      if (!query.state || !query.code) {
-        throw new Error(
-          '[@octokit/oauth-app] Both "code" & "state" parameters are required'
-        );
+      if (!query.code) {
+        throw new Error('[@octokit/oauth-app] "code" parameter is required');
```
[the remark this answers is not in the document]
Comment (baoshan) [reply src/middleware/handle-request.ts:93]: I think it is this issue: https://github.com/octokit/oauth-methods.js/issues/36

It is fixed by: https://github.com/octokit/oauth-methods.js/pull/52

[the remark this answers is not in the document]
Comment (baoshan) [reply src/middleware/handle-request.ts:93]: Please let me know if a separated PR is preferred.

[the remark this answers is not in the document]
Comment (gr2m) [reply src/middleware/handle-request.ts:93]: if you don't mind, that would be great. Than we can ship the fix as part of `3.x`, before we ship the rename of the middleware to `createWebWorkerHandler` which is a breaking change and will release 4.0.0

**Comment by baoshan** [comment]
I wish it is ready to be merged. Thanks @wolfy1339 and @gr2m for your help. ❤️

**Review by wolfy1339 — APPROVED** [review]

**Review by gr2m — DISMISSED** [review]
Thank you Baoshan!

File: src/index.ts [hunk 3]
```diff
@@ -9,49 +9,49 @@ import {
...
   OctokitTypeFromOptions,
-  ClientTypeFromOptions,
-  ClientType,
-  AddEventHandler,
+  Options,
   State,
 } from "./types";
 export { createNodeMiddleware } from "./middleware/node/index";
-export { createCloudflareHandler } from "./middleware/cloudflare/index";
+export { createWebWorkerHandler } from "./middleware/web-worker/index";
```
[the remark this answers is not in the document]
Comment (baoshan) [reply src/index.ts:54]: Great suggestion! Done.

**Review by gr2m — COMMENTED** [review]
Okay last thing: we should test that `createCloudflareHandler` is still working and that it logs out a deprecation message. I like to create a dedicated file to test deprecations: `test/deprecations.test.ts`. That way it's easy to remove all deprecations each time we release a breaking version.

Would you mind adding the file, testing that `createCloudflareHandler` still works as expected and that it logs the warning?

**Comment by baoshan** [comment]
Thanks for the suggestion! Done. Feel free to let me know if there is still room to improve.

**Review by wolfy1339 — APPROVED** [review]
Looks good

**Review by gr2m — APPROVED** [review]
great PR, thank you Baoshan!

**Pull request merged**